ops/pebble.py

```diff
@@ -88,9 +89,24 @@
                                 },
                                total=False)
 
-HttpDict = typing.TypedDict('HttpDict', {'url': str})
-TcpDict = typing.TypedDict('TcpDict', {'port': int})
-ExecDict = typing.TypedDict('ExecDict', {'command': str})
+HttpDict = typing.TypedDict('HttpDict',
+                            {'url': str,
+                             'headers': Dict[str, str]},
+                            total=False)
+TcpDict = typing.TypedDict('TcpDict',
+                           {'port': int,
+                            'host': str},
+                           total=False)
+ExecDict = typing.TypedDict('ExecDict',
+                            {'command': str,
+                             'service-context': str,
+                             'environment': Dict[str, str],
+                             'user-id': Optional[int],
+                             'user': str,
+                             'group-id': Optional[int],
+                             'group': str,
+                             'working-dir': str},
+                            total=False)
```

Comment:
These types only just became public (in ops 2.4.0), so that's probably the main reason. I wanted them to be public so people could use them to type check their `pebble.LayerDict`s in their charms. If people are using this new type annotations for their dicts, it will check them (though they don't have to).

Separately, we're planning to add a `--dry` mode for validating layer config to Pebble (https://github.com/canonical/pebble/pull/214). But I think additional work here is out of scope, at least for now.